scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -9,11 +9,12 @@
 
 from scrapy.utils.httpobj import urlparse_cached
 from scrapy.exceptions import NotConfigured
-
+from scrapy.utils.python import to_bytes
 
 class HttpProxyMiddleware(object):
 
-    def __init__(self):
+    def __init__(self, httproxy_encoding='latin-1'):
```

Comment:
IMHO it is better to have a default value in a single place (default_settings). 
What about renaming parameter to auth_encoding? It is clear this parameter is for http proxies because class name is HttpProxyMiddleware.